7_react/react1/week2/todo-app/src/todoList/TodoList.jsx

```diff
@@ -0,0 +1,53 @@
+import TimeCount from "../timer/Timer";
+import TodoItem from "../todoItem/TodoItem";
+import todos from "../todoItem/todos";
+import { useState } from "react";
+
+function AppTodo() {
+  // const data = [...todos];
+  const [todolist, setTodoList] = useState([...todos]);
+
+  const addTodo = () => {
+    const newTodo = {id:new Date(), name: "new todo", date: new Date(), status: false };
+    setTodoList((todolist) => { return [...todolist, newTodo]; });
+  }
+ 
+  const onDeleteClick = (id) => {
+    setTodoList(todolist.filter(todo => todo.id !== id));
+  }
+
+  const TodoList = (props) => {
```

Comment:
The react pattern is that every component has it's own file. In your case, the TodoList component is not reusable as it is declared inside the AppTodo component, and can only be called here. If that was intended, you could move lines 21-40 to line 48, and use it directly in the AppTodo JSX without creating a separate component. Otherwise, if you want the TodoList component to be reusable outside of AppTodo - create a separate file for it, and import the component into this one.